src/Symfony/Component/DependencyInjection/Definition.php

```diff
@@ -796,7 +800,7 @@ public function isDeprecated()
      */
     public function getDeprecationMessage($id)
     {
-        return str_replace('%service_id%', $id, $this->deprecationTemplate);
+        return null !== $this->deprecationTemplate ? str_replace('%service_id%', $id, $this->deprecationTemplate) : null;
```

Comment:
like I said in another comment, the point for me is to not dump garbage when doing `dump($definition)`